src/GovUk.Education.ExploreEducationStatistics.Publisher/Functions/NotifyChangeFunction.cs

```diff
@@ -48,25 +48,46 @@ public class NotifyChangeFunction
         {
             logger.LogInformation($"{executionContext.FunctionName} triggered: {message}");
             await MarkScheduledReleaseStatusAsSuperseded(message);
-            await ValidateReleaseAsync(message, async () =>
+            if (! await ValidatePublishingState(message, logger))
             {
-                if (message.Immediate)
+                await ValidateRelease(message, async () =>
                 {
-                    var releaseStatus = await CreateReleaseStatusAsync(message, ImmediateReleaseStartedState);
-                    await _queueService.QueuePublishReleaseFilesMessageAsync(releaseStatus.ReleaseId, releaseStatus.Id);
-                }
-                else
-                {
-                    await CreateReleaseStatusAsync(message, ScheduledState);
-                }
-            });
+                    if (message.Immediate)
+                    {
+                        var releaseStatus = await CreateReleaseStatusAsync(message, ImmediateReleaseStartedState);
+                        await _queueService.QueuePublishReleaseFilesMessageAsync(releaseStatus.ReleaseId,
+                            releaseStatus.Id);
+                    }
+                    else
+                    {
+                        await CreateReleaseStatusAsync(message, ScheduledState);
+                    }
+                });
+            }
+
             logger.LogInformation($"{executionContext.FunctionName} completed");
         }
 
-        private async Task ValidateReleaseAsync(NotifyChangeMessage message, Func<Task> andThen)
+        private async Task<bool> ValidatePublishingState(NotifyChangeMessage message, ILogger logger)
+        {
+            var (valid, logMessages) = 
+                await _validationService.ValidatePublishingState(message.ReleaseId);
+
+            if (!valid)
+            {
+                foreach (var releaseStatusLogMessage in logMessages)
```

Comment:
Do we log messages outside `ValidationService` so its easier to associate the error with the Azure function? Seems weird to be returning specific log messages back to the service consumer.

Not necessarily suggesting we change all of this, but wouldn't an `Either<ValidationError, bool>` be a better return type from `validationService.ValidatePublishingState`?

Reply:
We return them outside `ValidationService` because there is a messages field on the ReleaseStatus and we write that in one transaction when creating it.

Later on once the ReleaseStatus exists you can update a state with additional messages which are appended.

Originally we returned these messages to the UI but temporarily removed them from the API response now. I think it would be nice to include them again just for the BAU user. If we used the ApplicationInsights logger they wouldn't be as easily accessible.

Outside the ValidationService we know more about what the initial states of all the stages are going to be which depends on other stuff like whether it's immediate.  Either valid or invalid, we create the ReleaseStatus along with any messages.

I've now changed `ValidateRelease` to use Either as suggested and introduced `Unit` to represent a void return type, although this has removed the possibility of having successful validation with messages but that wasn't used anyway.

I've also changed `ValidatePublishingState` which happens earlier to do it's own and just return `bool`.

Reply:
Cool, didn't realise we were actually stuffing them into a messages field. There wouldn't be any danger of us running out of space for the table storage row if there are too many messages though right?

Nice work, think this makes a lot more sense now!

Reply:
There is a danger, but I'd assumed we wouldn't reach it if we are just sticking in a couple of validation messages or the summary of an exception from data factory.

According to https://docs.microsoft.com/en-us/rest/api/storageservices/Understanding-the-Table-Service-Data-Model a string field has a limit of 64KiB.

In UTF-8, if our messages only contained [A-Z0-9], that should give us 65536 characters at 1 byte per character. However the docs also say it's around 32K characters and I tested it with a string of random numbers and it's only allowing 32290.

I have created EES-1215 on making sure we don't exceed the limit.